test/e2e/node_feature_discovery_test.go

```diff
@@ -771,6 +790,68 @@ core:
 					Expect(err).NotTo(HaveOccurred())
 				})
 			})
+
+			Context("and check whether master config passed successfully or not", func() {
+				AfterEach(func() {
+					customMasterPodSpecOpts = nil
+				})
```

Comment:
This should be added into the "main" AfterEach() method instead. I.e. the one that where we cleanup the labels and CRs.